routing/dht/routing.go

```diff
@@ -125,17 +152,40 @@ func (dht *IpfsDHT) GetValue(ctx context.Context, key u.Key) ([]byte, error) {
 	})
 
 	// run it!
-	result, err := query.Run(ctx, rtp)
-	if err != nil {
+	_, err = query.Run(ctx, rtp)
+	if err != nil && len(vals) == 0 {
 		return nil, err
 	}
 
-	log.Debugf("GetValue %v %v", key, result.value)
-	if result.value == nil {
+	// process values retreived
+	best := dht.mostValid(vals)
```

Comment:
As described in https://github.com/ipfs/go-ipfs/pull/999#issuecomment-88603721 -- here may be a good place to defer to some user defined function:

```go
best := dht.Validator.MostValid(vals)
```